src/widget/checkbox.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A checkbox widget.
+
+use crate::kurbo::{BezPath, Point, RoundedRect, Size};
+use crate::piet::{LineCap, LineJoin, LinearGradient, RenderContext, StrokeStyle, UnitPoint};
+use crate::theme;
+use crate::widget::Align;
+use crate::{
+    Action, BaseState, BoxConstraints, Env, Event, EventCtx, LayoutCtx, PaintCtx, UpdateCtx, Widget,
+};
+
+/// A checkbox that toogles a boolean
```

Comment:
Not really qualified to make comments about the implementation, but noticed a typo here. 😄 
```suggestion
/// A checkbox that toggles a boolean
```

Reply:
Thank yoo for catching! Also: I love this suggestion thing.